cunumeric/deferred.py

```diff
@@ -3487,25 +3487,68 @@ def scan(
             assert self.shape == swapped.shape
             self.copy(swapped, deep=True)
 
-    def unique(self) -> NumPyThunk:
-        result = self.runtime.create_unbound_thunk(self.base.type)
-
+    def unique(
+        self, return_index: bool = False
+    ) -> Union[NumPyThunk, tuple[NumPyThunk, Optional[NumPyThunk]]]:
         task = self.context.create_auto_task(CuNumericOpCode.UNIQUE)
 
-        task.add_output(result.base)
         task.add_input(self.base)
+        task.add_scalar_arg(return_index, ty.bool_)
 
+        result = None
+        # Assuming legate core will always choose GPU variant
```

Comment:
[nit]: Add a comment as to why the branches here are creating the thunk in different ways (i.e. different types)

Reply:
Extra nit -- comments should be complete sentences, so punctuation and periods please.